Library/Homebrew/utils/repology.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require "net/http"
+require "json"
+
+module RepologyParser
+  def call_api(url)
+    puts "- Calling API #{url}"
+    uri = URI(url)
+    response = Net::HTTP.get(uri)
+
+    puts "- Parsing response"
+    JSON.parse(response)
+  end
+
+  def query_repology_api(last_package_in_response = "")
+    url = "https://repology.org/api/v1/projects/#{last_package_in_response}?inrepo=homebrew&outdated=1"
+
+    call_api(url)
+  end
+
+  def parse_repology_api
+    puts "\n-------- Query outdated packages from Repology --------"
+    page_no = 1
+    puts "\n- Paginating repology api page: #{page_no}"
+
+    outdated_packages = query_repology_api("")
+    last_pacakge_index = outdated_packages.size - 1
+    response_size = outdated_packages.size
+
+    while response_size > 1
```

Comment:
Is there anything in the API response that provides a number of pages?

Reply:
@MikeMcQuaid there's nothing in the API to paginate the response. The API limits the results to 200 packages per response and to paginate through the API, you have to use the last package in every response to get the next set of response till you have only one package in the response which indicates the last page.

Reply:
Would be good to have a maximum limit so this loop doesn't get "stuck" forever.

Reply:
Noted! @emctackett and I have agreed to limit it to a maximum of 15 paginations.